src/opendr/perception/facial_expression_recognition/image_based_facial_emotion_estimation/facial_emotion_learner.py

```diff
@@ -0,0 +1,799 @@
+        :type url: str, optional
+        """
+
+        valid_modes = ["data", "pretrained", "demo_image", "demo_video"]
+        if mode not in valid_modes:
+            raise UserWarning("mode parameter not valid:", mode, ", file should be one of:", valid_modes)
+        if path is None:
+            path = self.temp_path
+        if not os.path.exists(path):
+            os.makedirs(path)
+
+        if mode == "data":
+            print("Downloading data...")
+            data_path = os.path.join(path, 'data')
+            if not os.path.exists(data_path):
+                os.makedirs(data_path)
+
+            zip_path = os.path.join(path, 'data/AffectNet_micro.zip')
+            if not os.path.exists(zip_path):
+                # Download data
+                file_url = os.path.join(url, 'data/AffectNet_micro.zip')
+                urlretrieve(file_url, zip_path)
+                with zipfile.ZipFile(zip_path, 'r') as zip_ref:
+                    zip_ref.extractall(data_path)
+            else:
+                print("Data files already exist.")
+            print("Data download complete.")
+            downloaded_files_path = os.path.join(data_path, 'AffectNet_micro')
+
+        elif mode == "pretrained":
+            print("Downloading pretrained model weights...")
+            model_path = os.path.join(path, 'pretrained')
+            if not os.path.exists(model_path):
+                os.makedirs(model_path)
+
+            zip_path = os.path.join(path, 'pretrained/esr_9.zip')
+            if not os.path.exists(zip_path):
+                # Download data
+                file_url = os.path.join(url, 'pretrained/esr_9.zip')
+                urlretrieve(file_url, zip_path)
+                with zipfile.ZipFile(zip_path, 'r') as zip_ref:
+                    zip_ref.extractall(model_path)
+            else:
+                print("Pretrained files already exist.")
+            print("Pretrained model weights download complete.")
+            downloaded_files_path = os.path.join(model_path, 'esr_9')
+
+        elif mode == "demo_image":
+            print("Downloading image...")
+            demo_path = os.path.join(path, 'demo')
+            if not os.path.exists(demo_path):
+                os.makedirs(demo_path)
+
+            img_path = os.path.join(demo_path, 'sheldon.jpg')
+            if not os.path.exists(img_path):
+                # Download data
+                file_url = os.path.join(url, 'demo/sheldon.jpg')
+                urlretrieve(file_url, img_path)
+            else:
+                print("Data files already exist.")
+            print("Data download complete.")
+            downloaded_files_path = img_path
+
+        elif mode == "demo_video":
+            print("Downloading video...")
+            demo_path = os.path.join(path, 'demo')
+            if not os.path.exists(demo_path):
+                os.makedirs(demo_path)
+
+            vid_path = os.path.join(demo_path, 'sheldon.mp4')
```

Comment:
Typo in the filename, files should be up now.

```suggestion
            vid_path = os.path.join(demo_path, 'sheldon.jpg')
```

Reply:
sorry basically this is in the `demo_video` mode, it should probably set it to 'big_bang.mp4'